Assets/Scripts/Grid/Positioning/SpiralSequenceRandomPositionProvider.cs

```diff
@@ -0,0 +1,119 @@
+                    currentPosition.y = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+                
+                //If at this point the position is (center.x - maxDistance, center.y - maxDistance) the spiral is
+                //complete for maxDistance.
+                if (currentPosition == spiralEndTile) break;
+
+                stretchStart = new Vector2(currentPosition.x, currentPosition.y - 1);
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.x; i < stretchStart.x + k; i++) {
+                    currentPosition.x = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x + 1, currentPosition.y);
+                k++;
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.y; i < stretchStart.y + k; i++) {
+                    currentPosition.y = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x, currentPosition.y + 1);
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.x; i > stretchStart.x - k; i--) {
+                    currentPosition.x = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x - 1, currentPosition.y);
+                k++;
+            } while (true);
+
+            return ChooseRandomPositionsFromAvailableTiles(availableTiles, numTilesToGenerate);
+        }
+
+        private void AddIfAvailableTile(LinkedList<Vector2> availableTiles, IGrid grid, Vector2 tile) {
+            if (grid.TileBounds.Contains(tile)) {
+                availableTiles.AddLast(tile);
+                Debug.Log("Tile (" + tile.x + "," + tile.y + ") Chosen");
+            }
+        }
+
+        private Vector2[] ChooseRandomPositionsFromAvailableTiles(LinkedList<Vector2> availableTiles, 
+                                                                  int numTilesToGenerate) {
+
+            Vector2[] tiles = new Vector2[numTilesToGenerate];
+            int chosenTiles = 0;
+            float randomNumber;
+            
+            for (int i = 0; i < MaxTries; i++) {
+                if (chosenTiles == numTilesToGenerate) break;
+                
+                foreach (var tile in availableTiles) {
+                    randomNumber = _randomProvider.GetRandomFloatInRange(0, 1);
+                    if (randomNumber <= TileChooseProbability) {
+                        tiles[chosenTiles] = tile;
+                        chosenTiles++;
+                        availableTiles.Remove(tile);
```

Comment:
I suppose that the time complexity of removing an element is O(L), where L is the list size, as it has to do a linear search.

`ChooseRandomPositionsFromAvailableTiles` would be O(L<sup>2</sup>). Well, it usually would be O(L<sup>2</sup>*M) where M is MaxTries, but because in this case is a known constant, which is 100, wouldn't be considered.

But considering how this function works clustering the chosen tiles near the center of the spiral (at the beginning of the list) and because there's a break every time the tile is chosen, there would be a bunch of amortized time.